specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,538 @@
+    ///
+    /// The `mt_transfer_call` process:
+    ///
+    /// 1. Sender calls `mt_transfer_call` on MultiToken contract
+    /// 2. MultiToken contract transfers token from sender to receiver
+    /// 3. MultiToken contract calls `mt_on_transfer` on receiver contract
+    /// 4+. [receiver contract may make other cross-contract calls]
+    /// N. MultiToken contract resolves promise chain with `mt_resolve_transfer`, and may
+    ///    transfer token back to sender
+    ///
+    /// Requirements:
+    /// * Contract MUST forbid calls to this function by any account except self
+    /// * If promise chain failed, contract MUST revert token transfer
+    /// * If promise chain resolves with `true`, contract MUST return token to
+    ///   `sender_id`
+    ///
+    /// Arguments:
+    /// * `previous_owner_id`: the owner prior to the call to `mt_transfer_call`
+    /// * `receiver_id`: the `receiver_id` argument given to `mt_transfer_call`
+    /// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+    /// * `approvals`: if using Approval Management, contract MUST provide
+    ///   set of original approved accounts in this argument, and restore these
+    ///   approved accounts in case of revert. In this case it may be multiple sets of approvals
+    ///
+    /// Returns true if tokens were successfully transferred to `receiver_id`.
+    fn mt_resolve_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+    ) -> Vec<U128>;
+}
+```
+### Storage Management Trait
+#### Notes
+This is semi necessary for ft token types to be able to refund users for storage of many different token types like gold/silver... this might be slightly out of scope
+```
+pub trait StorageManagement {
+    // if `registration_only=true` MUST refund above the minimum balance if the account didn't exist and
+    //     refund full deposit if the account exists.
+    fn storage_deposit(
+        &mut self,
+        token_ids: Vec<TokenId>,
+        account_id: Option<AccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance;
+
+    /// Withdraw specified amount of available Ⓝ for predecessor account.
+    ///
+    /// This method is safe to call. It MUST NOT remove data.
+    ///
+    /// `amount` is sent as a string representing an unsigned 128-bit integer. If
+    /// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+    /// predecessor account's available balance, contract MUST panic.
+    ///
+    /// If predecessor account not registered, contract MUST panic.
+    ///
+    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted
+    /// function-call access-key call (UX wallet security)
+    ///
+    /// Returns the StorageBalance structure showing updated balances.
+    fn storage_withdraw(&mut self, token_ids:Vec<TokenId>, amount: Option<U128>) -> StorageBalance;
```

Comment:
Maybe 1 yoctoNEAR is not necessary in withdraw cause the token will go back to the account that key indicates, won't cause any loss for the user even if an access key is stolen.

Reply:
So for this is a preventative measure , to push users towards the wallet to make sure they don't get fished or do something they weren't intending to do. Perhaps we can do an unsafe for applications that need this in the future.